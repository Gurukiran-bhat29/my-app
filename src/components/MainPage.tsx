import { useState, useEffect } from 'react';
import { Link } from "react-router-dom";
import { Article } from '../utils/article';
import { infoUrl } from '../utils/constants';
import InfoCard from './InfoCard';
import ErrorBoundary from './ErrorBoundary';

function MainPage() {
  const [articles, setArticles] = useState([]);
  const [error, setError] = useState(false);

  const getInfo = async () => {
    try {
      const res = await fetch(infoUrl);

      if (!res.ok) {
        setError(true);
      }

      const jsonRes = await res.json();
      setArticles(jsonRes);
    } catch (error) {
      console.log('Error: ', error);
    }
  }

  useEffect(() => {
    getInfo();
  }, [])

  if (error) {
    return <ErrorBoundary />
  }

  return (
    <div className='main-container'>
      {articles.length > 0 && articles.map(({ id, title, summary }: Article) => {
        return (
          <Link data-testid='info-cards' className='info-card' key={id} to={'results?v=' + id}>
            <InfoCard title={title} summary={summary} />
          </Link>
        )
      })}
    </div>
  );
}

export default MainPage;
