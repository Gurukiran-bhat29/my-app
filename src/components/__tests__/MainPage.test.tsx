import { render, screen, waitFor } from "@testing-library/react"
import MainPage from "../MainPage"
import { BrowserRouter } from "react-router-dom";
import MOCK_DATA from '../../MockData/articlesData.json'

beforeEach(() => {
  jest.clearAllMocks();
  global.fetch = jest.fn(() =>
    Promise.resolve({
      ok: true,
      json: () =>
        Promise.resolve(MOCK_DATA),
    })
  ) as jest.Mock;
});

test('should render articles list on load', async () => {
  const articleContainer = render(
    <BrowserRouter>
      <MainPage />
    </BrowserRouter>
  );
  
  await waitFor(() => expect(screen.getAllByTestId('info-cards').length).toBe(10));

  expect(articleContainer).toMatchSnapshot();
})
