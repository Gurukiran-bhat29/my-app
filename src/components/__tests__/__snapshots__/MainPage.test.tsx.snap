// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render article on load 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-container"
      >
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=1"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              Exploring the Wonders of the Great Barrier Reef
            </p>
            <p>
              Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=10"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The Impact of Artificial Intelligence on Society: Opportunities and Challenges
            </p>
            <p>
              Examine the transformative effects of artificial intelligence on various aspects of society and discuss ethical considerations.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=2"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The Rise of Electric Vehicles: Shaping the Future of Transportation
            </p>
            <p>
              Learn about the advancements in electric vehicle technology and their impact on reducing carbon emissions.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=3"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The Art of Meditation: Finding Inner Peace in a Busy World
            </p>
            <p>
              Explore the benefits of meditation for mental health and wellbeing.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=4"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              Unraveling the Mysteries of Black Holes
            </p>
            <p>
              Delve into the fascinating world of black holes and their role in shaping the universe.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=5"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The Power of Habit: How Small Changes Can Transform Your Life
            </p>
            <p>
              Learn about the science behind habits and strategies for cultivating positive ones.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=6"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              Culinary Adventures: Exploring the Flavors of Italy
            </p>
            <p>
              Embark on a gastronomic journey through Italy's rich culinary heritage.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=7"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The Human Brain: Unlocking Its Potential and Understanding Its Complexity
            </p>
            <p>
              Discover the intricacies of the human brain and ongoing research aimed at unraveling its mysteries.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=8"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              Sustainable Living: Simple Steps Towards a Greener Future
            </p>
            <p>
              Explore practical tips and eco-friendly practices for reducing carbon footprint.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=9"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The History and Legacy of the Ancient Egyptians
            </p>
            <p>
              Journey back in time to ancient Egypt and uncover the marvels of its civilization.
            </p>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-container"
    >
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=1"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            Exploring the Wonders of the Great Barrier Reef
          </p>
          <p>
            Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=10"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The Impact of Artificial Intelligence on Society: Opportunities and Challenges
          </p>
          <p>
            Examine the transformative effects of artificial intelligence on various aspects of society and discuss ethical considerations.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=2"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The Rise of Electric Vehicles: Shaping the Future of Transportation
          </p>
          <p>
            Learn about the advancements in electric vehicle technology and their impact on reducing carbon emissions.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=3"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The Art of Meditation: Finding Inner Peace in a Busy World
          </p>
          <p>
            Explore the benefits of meditation for mental health and wellbeing.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=4"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            Unraveling the Mysteries of Black Holes
          </p>
          <p>
            Delve into the fascinating world of black holes and their role in shaping the universe.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=5"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The Power of Habit: How Small Changes Can Transform Your Life
          </p>
          <p>
            Learn about the science behind habits and strategies for cultivating positive ones.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=6"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            Culinary Adventures: Exploring the Flavors of Italy
          </p>
          <p>
            Embark on a gastronomic journey through Italy's rich culinary heritage.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=7"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The Human Brain: Unlocking Its Potential and Understanding Its Complexity
          </p>
          <p>
            Discover the intricacies of the human brain and ongoing research aimed at unraveling its mysteries.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=8"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            Sustainable Living: Simple Steps Towards a Greener Future
          </p>
          <p>
            Explore practical tips and eco-friendly practices for reducing carbon footprint.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=9"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The History and Legacy of the Ancient Egyptians
          </p>
          <p>
            Journey back in time to ancient Egypt and uncover the marvels of its civilization.
          </p>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render articles list on load 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-container"
      >
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=1"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              Exploring the Wonders of the Great Barrier Reef
            </p>
            <p>
              Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=10"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The Impact of Artificial Intelligence on Society: Opportunities and Challenges
            </p>
            <p>
              Examine the transformative effects of artificial intelligence on various aspects of society and discuss ethical considerations.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=2"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The Rise of Electric Vehicles: Shaping the Future of Transportation
            </p>
            <p>
              Learn about the advancements in electric vehicle technology and their impact on reducing carbon emissions.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=3"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The Art of Meditation: Finding Inner Peace in a Busy World
            </p>
            <p>
              Explore the benefits of meditation for mental health and wellbeing.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=4"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              Unraveling the Mysteries of Black Holes
            </p>
            <p>
              Delve into the fascinating world of black holes and their role in shaping the universe.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=5"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The Power of Habit: How Small Changes Can Transform Your Life
            </p>
            <p>
              Learn about the science behind habits and strategies for cultivating positive ones.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=6"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              Culinary Adventures: Exploring the Flavors of Italy
            </p>
            <p>
              Embark on a gastronomic journey through Italy's rich culinary heritage.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=7"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The Human Brain: Unlocking Its Potential and Understanding Its Complexity
            </p>
            <p>
              Discover the intricacies of the human brain and ongoing research aimed at unraveling its mysteries.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=8"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              Sustainable Living: Simple Steps Towards a Greener Future
            </p>
            <p>
              Explore practical tips and eco-friendly practices for reducing carbon footprint.
            </p>
          </div>
        </a>
        <a
          class="info-card"
          data-testid="info-cards"
          href="/results?v=9"
        >
          <div
            class="container"
          >
            <p
              class="title"
            >
              The History and Legacy of the Ancient Egyptians
            </p>
            <p>
              Journey back in time to ancient Egypt and uncover the marvels of its civilization.
            </p>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-container"
    >
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=1"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            Exploring the Wonders of the Great Barrier Reef
          </p>
          <p>
            Discover the incredible biodiversity and natural beauty of the Great Barrier Reef.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=10"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The Impact of Artificial Intelligence on Society: Opportunities and Challenges
          </p>
          <p>
            Examine the transformative effects of artificial intelligence on various aspects of society and discuss ethical considerations.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=2"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The Rise of Electric Vehicles: Shaping the Future of Transportation
          </p>
          <p>
            Learn about the advancements in electric vehicle technology and their impact on reducing carbon emissions.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=3"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The Art of Meditation: Finding Inner Peace in a Busy World
          </p>
          <p>
            Explore the benefits of meditation for mental health and wellbeing.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=4"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            Unraveling the Mysteries of Black Holes
          </p>
          <p>
            Delve into the fascinating world of black holes and their role in shaping the universe.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=5"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The Power of Habit: How Small Changes Can Transform Your Life
          </p>
          <p>
            Learn about the science behind habits and strategies for cultivating positive ones.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=6"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            Culinary Adventures: Exploring the Flavors of Italy
          </p>
          <p>
            Embark on a gastronomic journey through Italy's rich culinary heritage.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=7"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The Human Brain: Unlocking Its Potential and Understanding Its Complexity
          </p>
          <p>
            Discover the intricacies of the human brain and ongoing research aimed at unraveling its mysteries.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=8"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            Sustainable Living: Simple Steps Towards a Greener Future
          </p>
          <p>
            Explore practical tips and eco-friendly practices for reducing carbon footprint.
          </p>
        </div>
      </a>
      <a
        class="info-card"
        data-testid="info-cards"
        href="/results?v=9"
      >
        <div
          class="container"
        >
          <p
            class="title"
          >
            The History and Legacy of the Ancient Egyptians
          </p>
          <p>
            Journey back in time to ancient Egypt and uncover the marvels of its civilization.
          </p>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
